src/tests/roc_core/test_heap_arena.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_core/heap_arena.h"
+#include "roc_core/memory_ops.h"
+
+namespace roc {
+namespace core {
+
+TEST_GROUP(heap_arena) {
+    // clang-format off
+    void setup() {
+        core::HeapArena::set_flags(core::DefaultHeapArenaFlags
+                                    & ~core::HeapArenaFlag_EnableGuards);
+    }
+    void teardown() {
+        core::HeapArena::set_flags(core::DefaultHeapArenaFlags
+                                    | core::HeapArenaFlag_EnableGuards);
+    }
+// clang-format on
+}; // namespace roc
```

Comment:
```suggestion
// clang-format off
TEST_GROUP(heap_arena) {
    void setup() {
        core::HeapArena::set_flags(core::DefaultHeapArenaFlags
                                    & ~core::HeapArenaFlag_EnableGuards);
    }
    void teardown() {
        core::HeapArena::set_flags(core::DefaultHeapArenaFlags);
    }
};
// clang-format on
```

Reply:
* Formatting is still broken. Here is a fix.
* In teardown(), we can use just DefaultHeapArenaFlags, without `| core::HeapArenaFlag_EnableGuards` part.

Reply:
Yes. I should have done that.